File: client/viewmodels/Customer.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace viewmodels
{
    public class Customer
    {
        public Guid? Id { get; set; }
        public int Version { get; set; }
        public string? Status { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime UpdatedDate { get; set; }
        public string Name { get; set; }
        public string? DeliveryAddress { get; set; }
        public string? DeliveryCity { get; set; }
        public string? DeliveryPostalCode { get; set; }
        public bool Active { get; set; }
        public List<Order>? Orders { get; set; } = new List<Order>();

    }
}

File: client/frontend/src/app/shared/models/customer.ts
import { Order } from "./order";

export class Customer {
    id: string | undefined;
    version: number | undefined;
    status: string | undefined;
    createDate: Date | undefined;
    updateDate: Date | undefined;
    name: string;
    deliveryAddress: string | undefined;
    deliveryCity: string | undefined;
    deliveryPostalCode: string | undefined;
    active: boolean;
    orders: Order[] | undefined;

    constructor(id: string | undefined, version: number | undefined, status: string | undefined, createDate: Date | undefined, updateDate: Date | undefined, name: string, deliveryAddress: string | undefined, deliveryCity: string | undefined, deliveryPostalCode: string | undefined, active: boolean, orders: Order[] | undefined){
        this.id = id;
        this.version = version;
        this.status = status;
        this.createDate = createDate;
        this.updateDate = updateDate;
        this.name = name;
        this.deliveryAddress = deliveryAddress;
        this.deliveryCity = deliveryCity;
        this.deliveryPostalCode = deliveryPostalCode;
        this.active = active;
        this.orders = orders;
    }

    static createNew(name: string, deliveryAddress: string, deliveryCity: string, deliveryPostalCode: string): Customer {
        return new Customer(undefined, undefined, undefined, undefined, undefined, name, deliveryAddress, deliveryCity, deliveryPostalCode, true, undefined);
    }
}